pkg/check/postage/postage.go

```diff
@@ -0,0 +1,190 @@
+		PostageLabel:       "test-label",
+		PostageWait:        5 * time.Second,
+		NodeCount:          1,
+	}
+}
+
+// compile check whether Check implements interface
+var _ beekeeper.Action = (*Check)(nil)
+
+// Check instance
+type Check struct{}
+
+// NewCheck returns new check
+func NewCheck() beekeeper.Action {
+	return &Check{}
+}
+
+func (c *Check) Run(ctx context.Context, cluster *bee.Cluster, opts interface{}) (err error) {
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+
+	sortedNodes := cluster.NodeNames()
+	node := sortedNodes[0]
+
+	client := clients[node]
+
+	batchID, err := client.CreatePostageBatch(ctx, o.PostageAmount, o.PostageDepth, o.GasPrice, o.PostageLabel, false)
+	if err != nil {
+		return fmt.Errorf("node %s: batch id %w", node, err)
+	}
+	time.Sleep(o.PostageWait)
+
+	batches, err := client.PostageBatches(ctx)
+	if err != nil {
+		return fmt.Errorf("failed getting postage batches %w", err)
+	}
+
+	found := false
+	for _, v := range batches {
+		if v.BatchID == batchID {
+			found = true
+			if v.Amount.Int64() != o.PostageAmount {
+				return fmt.Errorf(
+					"invalid batch amount expected %d got %d, batch %s",
+					o.PostageAmount,
+					v.Amount.Int64(),
+					batchID,
+				)
+			}
+			if v.Depth != uint8(o.PostageDepth) {
+				return fmt.Errorf(
+					"invalid batch amount expected %d got %d, batch %s",
+					o.PostageDepth,
+					v.Depth,
+					batchID,
+				)
+			}
+		}
+	}
+	if !found {
+		return fmt.Errorf("cannot find batch %s", batchID)
+	}
+
+	fmt.Printf("node %s: created new batch id %s\n", node, batchID)
+
+	err = client.TopUpPostageBatch(ctx, batchID, o.PostageTopupAmount, o.GasPrice)
+	if err != nil {
+		return fmt.Errorf("failed topping up batch %s with amount %d gas %s: %w", batchID, o.PostageTopupAmount, o.GasPrice, err)
+	}
+	time.Sleep(o.PostageWait)
+
+	batches, err = client.PostageBatches(ctx)
+	if err != nil {
+		return fmt.Errorf("failed getting postage batches %w", err)
+	}
+
+	newAmount := o.PostageAmount + o.PostageTopupAmount
+
+	found = false
+	for _, v := range batches {
+		if v.BatchID == batchID {
+			found = true
+			if v.Amount.Int64() != newAmount {
+				return fmt.Errorf(
+					"invalid batch amount expected %d got %d, batch %s",
+					newAmount,
+					v.Amount.Int64(),
+					batchID,
+				)
+			}
+			if v.Depth != uint8(o.PostageDepth) {
+				return fmt.Errorf(
+					"invalid batch amount expected %d got %d, batch %s",
+					o.PostageDepth,
+					v.Depth,
+					batchID,
+				)
+			}
+		}
+	}
+	if !found {
+		return fmt.Errorf("cannot find batch %s", batchID)
+	}
+
+	fmt.Printf("node %s: topped up batch id %s\n", node, batchID)
+
+	depthChange := o.PostageNewDepth - o.PostageDepth
+
+	newValue2 := mbig.NewInt(0).Div(mbig.NewInt(newAmount), mbig.NewInt(int64(1<<depthChange)))
+
+	err = client.DilutePostageBatch(ctx, batchID, o.PostageNewDepth, o.GasPrice)
```

Comment:
consider in both API wrappers to actually wait for the dilution to get picket up (by polling)